### PR TITLE
feat: [DHIS2] New tracker endpoints in capture

### PR DESCRIPTION
**Pull request by eirikhaugstulen**
Temporary PR to collect epic of replacing the tracker endpoints

### CODE REVIEW

File: src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/NewEventWorkspace/NewEventWorkspace.component.js [hunk 1]
```diff
@@ -33,7 +33,8 @@ const NewEventWorkspacePlain = ({
     ...passOnProps
 }: Props) => {
     const selectedTab = useSelector(({ router: { location } }) => location.query.tab);
-    const { events, enrollmentDate, incidentDate } = useSelector(({ enrollmentDomain }) => enrollmentDomain?.enrollment);
+    const { events, enrolledAt, occurredAt } = useSelector(({ enrollmentDomain }) => enrollmentDomain?.enrollment);
+    console.log(enrolledAt, occurredAt);
```
Comment (Bekkalizer) [inline src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/NewEventWorkspace/NewEventWorkspace.component.js:37]: Should this be removed?

File: src/core_modules/capture-core/components/Pages/ViewEvent/Notes/viewEventNotes.epics.js [hunk 2]
```diff
@@ -53,20 +55,34 @@ export const addNoteForViewEventEpic = (action$: InputObservable, store: ReduxSt
         map((action) => {
             const state = store.value;
             const payload = action.payload;
-
             const eventId = state.viewEventPage.eventId;
+            const eventContainer = state.viewEventPage.loadedValues.eventContainer;
+            const { event: clientMainValues, values: clientValues } = eventContainer;
```
Comment (Bekkalizer) [inline src/core_modules/capture-core/components/Pages/ViewEvent/Notes/viewEventNotes.epics.js:60]: Values might not be needed when updating notes

File: src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPageDefault/EnrollmentAddEventPageDefault.container.js [hunk 3]
```diff
@@ -52,13 +52,15 @@ export const EnrollmentAddEventPageDefault = ({
     const selectedProgramStage = [...program.stages.values()].find(item => item.id === stageId);
     const outputEffects = useWidgetDataFromStore(widgetReducerName);
     const hideWidgets = useHideWidgetByRuleLocations(program.programRules);
+    // $FlowFixMe
+    const trackedEntityName = program?.trackedEntityType?.name;
```
Comment (eirikhaugstulen) [inline src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPageDefault/EnrollmentAddEventPageDefault.container.js:56]: This seems like a dirty hack from my side. Any ideas on any better solutions? Enrollment does no longer contain the TET.

**Comment by eirikhaugstulen** [comment]
Hey all,
it was decided that we should try to push this issue through to not block any other issues in this MS. Would be great if you could have a look / give it a test!

Known issues this far:
- Assigned user is not showing up in workinglists & viewEvent page. This is a BE-bug ([DHIS2-12620](https://jira.dhis2.org/browse/DHIS2-12620))
- Name of user that added notes is not displayed in the enrollment pages. This is a BE-bug ([DHIS2-12622](https://jira.dhis2.org/browse/DHIS2-12622))
- Relationships regarding single events is queried through old endpoints,  but posts through the new endpoints ([DHIS2-12625](https://jira.dhis2.org/browse/DHIS2-12625))
- Notes is fetched through the new endpoints, but posted to the old endpoints ([DHIS2-12622](https://jira.dhis2.org/browse/DHIS2-12622))

File: src/core_modules/capture-core/components/WidgetEnrollmentComment/WidgetEnrollmentComment.actions.js [hunk 4]
```diff
@@ -22,7 +22,7 @@ export const startAddNoteForEnrollment = (enrollmentId: string, serverData: Obje
     actionCreator(actionTypes.START_ADD_NOTE_FOR_ENROLLMENT)({ selections, context }, {
         offline: {
             effect: {
-                url: `enrollments/${enrollmentId}/note`,
+                url: 'tracker?async=false&importStrategy=UPDATE',
```
Comment (JoakimSM) [inline src/core_modules/capture-core/components/WidgetEnrollmentComment/WidgetEnrollmentComment.actions.js:25]: Should probably use the old endpoint here?

**Review by JoakimSM — COMMENTED** [review]
Good work on this @eirikhaugstulen! 👍

**Comment by eirikhaugstulen** [comment]
Hey @JoakimSM,
We were missing some functionality for caching and displaying scheduledAt label (dueDateLabel), but it should be added now. Please take a look at my latest commit and check if this is what you had in mind.
Thanks!

File: src/core_modules/capture-core/metaDataMemoryStoreBuilders/programs/factory/programStage/ProgramStageFactory.js [hunk 5]
```diff
@@ -131,7 +131,8 @@ export class ProgramStageFactory {
                 _form.description = cachedProgramStage.description;
                 _form.featureType = ProgramStageFactory._getFeatureType(cachedProgramStage);
                 _form.access = cachedProgramStage.access;
-                _form.addLabel({ id: 'eventDate', label: cachedProgramStage.executionDateLabel || 'Incident date' });
+                _form.addLabel({ id: 'occurredAt', label: cachedProgramStage.executionDateLabel || 'Incident date' });
+                _form.addLabel({ id: 'scheduledAt', label: cachedProgramStage.dueDateLabel || 'Scheduled at' });
```
Comment (JoakimSM) [inline src/core_modules/capture-core/metaDataMemoryStoreBuilders/programs/factory/programStage/ProgramStageFactory.js:135]: NIT suggestion
```suggestion
                _form.addLabel({ id: 'scheduledAt', label: cachedProgramStage.dueDateLabel || 'Scheduled date' });
```

**Review by JoakimSM — APPROVED** [review]
🎉

**Comment by simonadomnisoru** [comment]
👏

**Pull request merged**